design-system/react/src/components/InputAmount/InputAmount.tsx

```diff
@@ -1,28 +1,31 @@
-import { DECIMAL_UNITS } from '@fuel-ts/math';
 import type { BN } from '@fuel-ts/math';
+import { bn, DECIMAL_UNITS } from '@fuel-ts/math';
 import { cssObj } from '@fuel-ui/css';
 import { useEffect, useState } from 'react';
 import type { FC } from 'react';
 
 import { Button } from '../Button';
 import { Flex } from '../Flex';
+import type { InputProps } from '../Input';
 import { Input } from '../Input';
 import type { InputNumberProps } from '../Input/InputNumber';
 import { Text } from '../Text';
 
 import { InputAmountLoader } from './InputAmountLoader';
-import { createAmount } from './utils';
+import { createAmount, formatAmount } from './utils';
 
-export type InputAmountProps = {
+export type InputAmountProps = InputProps & {
   name?: string;
   balance?: BN;
+  balancePrecision?: number;
   value?: BN | null;
   units?: number;
   onChange?: (val: BN) => void;
   hiddenMaxButton?: boolean;
   hiddenBalance?: boolean;
-  max?: BN;
   inputProps?: InputNumberProps;
+  isDisabled?: boolean;
+  fee?: BN | null;
```

Comment:
The balance of the user fee should be informed on the application level. In this way, the user knows the max he can apply. Making it only when clicking on max makes this a hidden action for the user, that will not know what is happening.

Also the balance shown on the input should show the entire number of decimals, or we can add a tooltip when the user hovers to show the total balance.

Reply:
✅